client/events.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0
+//
+// SPDX-License-Identifier: EPL-2.0
+
+package client
+
+import (
+	"sync"
+)
+
+// EventsQueue is a bounded cyclic events queue, safe for concurrent use.
+// When capacity is reached, oldest events in the queue are replaced by newer ones.
+type EventsQueue struct {
```

Comment:
* Why this struct is called EventsQueue? It doesn't work with events but generic interface elements as the ring buffer.
* Even more, it seems that it's more than queue because the element consumption is there. Isn't it a status consumer?